docs/standard/serialization/binaryformatter-migration-guide/migrate-to-system-text-json.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "BinaryFormatter migration guide: Migrate to System.Text.Json (JSON)"
+description: "Migrate from BinaryFormatter to System.Text.Json for JSON serialization."
+ms.date: 5/31/2024
+no-loc: [BinaryFormatter, Serialization, System.Text.Json]
+dev_langs:
+  - CSharp
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "System.Text.Json"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# Migrate to System.Text.Json (JSON)
+
+[System.Text.Json](../system-text-json/overview.md) defaults to emphasizing literal, deterministic behavior and avoids any guessing or interpretation on the caller's behalf. The library is intentionally designed this way for security and performance. From the migration perspective, it's crucial to know the following facts:
+
+- By default, **fields are not serialized or deserialized**, but they can be [annotated for serialization](../system-text-json/fields.md), or `JsonSerializerOptions.IncludeFields` can be cautiously set to `true` to include all fields.
+
+  ```csharp
+  JsonSerializerOptions options = new()
+  {
+      IncludeFields = true
+  };
+  ```
+
+- By default, System.Text.Json **ignores private fields and properties**. You can enable use of a non-public accessor on a property by using the `[JsonInclude]` attribute. Including private fields requires some [non-trivial extra work](../system-text-json/custom-contracts.md#example-serialize-private-fields).
+
+- System.Text.Json **[cannot deserialize read-only fields](/dotnet/api/system.text.json.jsonserializeroptions.ignorereadonlyfields?view#remarks)** or properties, but the `[JsonConstructor]` attribute can be used to indicate that the specified constructor should be used to create instances of the type on deserialization. The constructor can set the read-only fields and properties.
+
+- It [supports serialization and deserialization of most built-in collections](../system-text-json/supported-collection-types.md). The exceptions:
+  - multi-dimensional arrays,
+  - `BitArray`,
+  - `LinkedList<T>`,
+  - `Dictionary<TKey, TValue>`, where `TKey` is not a primitive type,
+  - `BlockingCollection<T>` and `ConcurrentBag<T>`,
+  - most of the collections from [System.Collections.Specialized](../system-text-json/supported-collection-types.md#systemcollectionsspecialized-namespace) and [System.Collections.ObjectModel](../system-text-json/supported-collection-types.md#systemcollectionsobjectmodel-namespace) namespaces.
```

Comment:
This is far from an exhaustive list, user-defined collections types that have no default constructor and are not mutable (i.e. by implementing `ICollection<T>`) are also not supported. We also do not support some of the built-in collection interfaces (`IReadOnlySet<T>` jumps to mind).

Given that this list is changing every year, I would recommend deferring to STJ documentation (I believe it has a section noting supported types) which should be kept up to date as the library evolves.